c3d/c3d.py

```diff
@@ -2206,11 +2206,32 @@ def add_frames(self, frames, index=None):
             Insert the frame or sequence at the index (the first sequence frame will be inserted at give index).
             Note that the index should be relative to 0 rather then the frame number provided by read_frames()!
         '''
-        sh = np.array(frames, dtype=object).shape
+        # Determine shape without using np.array() which fails on inhomogeneous data
```

Comment:
This is the only thing I am not 100% sure about. Why was this change required in the first place? So what changed either in numpy or the c3d format that np.array on frames failed? Do you know @davidpagnon ?

Rest of the changes look good :)